streaming/base/util.py

```diff
@@ -399,21 +425,11 @@ def not_merged_index(index_file_path: str, out: str):
     local_index_files = []
     cl = CloudUploader.get(cu.local, exist_ok=True, keep_local=True)
     for file in cl.list_objects():
-        if file.endswith('.json') and not_merged_index(file, cu.local):
+        if file.endswith('.json') and _not_merged_index(file, cu.local):
             local_index_files.append(file)
 
     if cu.remote:
-        obj = urllib.parse.urlparse(cu.remote)
-        remote_index_files = []
-        for file in cu.list_objects():
-            if file.endswith(get_index_basename()) and not_merged_index(file, cu.remote):
-                join_char = '//'
-                if obj.scheme == 'dbfs':
-                    path = Path(cu.remote)
-                    prefix = os.path.join(path.parts[0], path.parts[1])
-                    if prefix == 'dbfs:/Volumes':
-                        join_char = '/'
-                remote_index_files.append(obj.scheme + join_char + os.path.join(obj.netloc, file))
+        remote_index_files = _format_remote_index_files(cu.list_objects(), cu.remote)
```

Comment:
Seems that this line like should be this?
```
        remote_index_files = _format_remote_index_files(cu.remote, cu.list_objects())
```

Reply:
You are right my bad 🤦, should be catch with E2E test `merge_index `

Reply:
No worries, I made a quick PR to fix this https://github.com/mosaicml/streaming/pull/584